pkg/controllers/repositoryserver/repositoryserver_controller.go

```diff
@@ -71,37 +71,66 @@ func (r *RepositoryServerReconciler) Reconcile(ctx context.Context, req ctrl.Req
 		return ctrl.Result{}, errors.Wrap(err, "Failed to get a k8s client")
 	}
 
-	logger.Info("Fetch RepositoryServer CR. If not found end reconcile loop")
 	repositoryServer := &crkanisteriov1alpha1.RepositoryServer{}
 	if err = r.Get(ctx, req.NamespacedName, repositoryServer); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	logger.Info("Setting the CR status as 'Pending' since a create or update event is in progress")
 	repositoryServer.Status.Progress = crkanisteriov1alpha1.Pending
 
-	logger.Info("Found RepositoryServer CR. Create or update owned resources")
 	repoServerHandler := newRepositoryServerHandler(ctx, req, logger, r, kubeCli, repositoryServer)
 	repoServerHandler.RepositoryServer = repositoryServer
 
 	if err = r.Status().Update(ctx, repoServerHandler.RepositoryServer); err != nil {
 		return ctrl.Result{}, err
 	}
+
+	logger.Info("Create or update owned resources")
 	if err := repoServerHandler.CreateOrUpdateOwnedResources(ctx); err != nil {
 		logger.Info("Setting the CR status as 'Failed' since an error occurred in create/update event")
-		repoServerHandler.RepositoryServer.Status.Progress = crkanisteriov1alpha1.Failed
-		if uerr := r.Status().Update(ctx, repoServerHandler.RepositoryServer); uerr != nil {
+		if uerr := repoServerHandler.updateProgressInCRStatus(ctx, crkanisteriov1alpha1.Failed); uerr != nil {
 			return ctrl.Result{}, uerr
 		}
 		r.Recorder.Event(repoServerHandler.RepositoryServer, corev1.EventTypeWarning, "Failed", err.Error())
 		return ctrl.Result{}, err
 	}
-	logger.Info("Setting the CR status as 'Ready' after completing the create/update event\n\n\n\n")
-	repoServerHandler.RepositoryServer.Status.Progress = crkanisteriov1alpha1.Ready
-	if err = r.Status().Update(ctx, repoServerHandler.RepositoryServer); err != nil {
+
+	logger.Info("Connect to Kopia Repository")
```

Comment:
Can you please share an example of how this log is displayed when the controller is running? Does it log the name and namespace fields? Should we also log

Reply:
`{"level":"debug","ts":"2023-07-20T16:17:29+05:30","msg":"Connect to Kopia Repository","controller":"repositoryserver","controllerGroup":"cr.kanister.io","controllerKind":"RepositoryServer","RepositoryServer":{"name":"kopia-repo-server-1","namespace":"kanister"},"namespace":"kanister","name":"kopia-repo-server-1","reconcileID":"b6b4d6f3-4cb6-4355-8f52-548ebace6bab"}`